docs: replace torch.distributed.run by torchrun

panpan0000 opened the pull request:
# What does this PR do?

The old way will be deprecated , move to new torchrun.
`FutureWarning: The module torch.distributed.launch is deprecated and will be removed in future. Use torchrun.`

This PR only addresses the doc part, not touching the unit-test to limit the impact.


<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

Fixes # (issue)


## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- text models: @ArthurZucker and @younesbelkada
- vision models: @amyeroberts
- speech models: @sanchit-gandhi
- graph models: @clefourrier

Library:

- flax: @sanchit-gandhi
- generate: @gante
- pipelines: @Narsil
- tensorflow: @gante and @Rocketknight1
- tokenizers: @ArthurZucker
- trainer: @muellerzr and @pacman100

Integrations:

- deepspeed: HF Trainer/Accelerate: @pacman100
- ray/raytune: @richardliaw, @amogkam
- Big Model Inference: @SunMarc
- quantization (bitsandbytes, autogpt): @SunMarc and @younesbelkada

Documentation: @stevhliu and @MKhalusova

HF projects:

- accelerate: [different repo](https://github.com/huggingface/accelerate)
- datasets: [different repo](https://github.com/huggingface/datasets)
- diffusers: [different repo](https://github.com/huggingface/diffusers)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Maintained examples (not research project or legacy):

- Flax: @sanchit-gandhi
- PyTorch: See Models above and tag the person corresponding to the modality of the example.
- TensorFlow: @Rocketknight1

 -->

panpan0000 commented:
I just found an old PR which did the similar work  https://github.com/huggingface/transformers/pull/21780/files but out of sync.
but my PR just focus on DOC

amyeroberts reviewed (approved):
Thanks for adding this! 

Just a few comments on removing the changes for the research examples. 

For any future readers who stumble on this PR: the previously closed PR requested [comments in the docs](https://github.com/huggingface/transformers/pull/21780#pullrequestreview-1315022415) for older pytorch versions. We now officially support pytorch >= 1.10. The entrypoint `torchrun` is present from [1.10 onwards](https://github.com/pytorch/pytorch/commit/65e6194aeb3269a182cfe2c05c122159da12770f).

panpan0000 commented on examples/research_projects/distillation/README.md:
got it . thank you @amyeroberts

panpan0000 committed:
docs: replace torch.distributed.run by torchrun

 `transformers` now officially support pytorch >= 1.10.
 The entrypoint `torchrun`` is present from 1.10 onwards.

Signed-off-by: Peter Pan <Peter.Pan@daocloud.io>

panpan0000 commented:
> Thanks for adding this!
> 
> Just a few comments on removing the changes for the research examples.
> 
> For any future readers who stumble on this PR: the previously closed PR requested [comments in the docs](https://github.com/huggingface/transformers/pull/21780#pullrequestreview-1315022415) for older pytorch versions. We now officially support pytorch >= 1.10. The entrypoint `torchrun` is present from [1.10 onwards](https://github.com/pytorch/pytorch/commit/65e6194aeb3269a182cfe2c05c122159da12770f).

comments already addressed, now PR can be merged now.

ArthurZucker commented on src/transformers/trainer.py:
```diff
@@ -1621,7 +1621,7 @@ def _inner_training_loop(
                 raise ValueError(
                     "Currently --debug underflow_overflow is not supported under DP. Please use DDP"
-                    " (torch.distributed.launch)."
+                    " (torch.distributed.launch or torchrun)."
```
```suggestion
                    " (torchrun or torch.distributed.launch (deprecated))."
```

panpan0000 committed:
Update src/transformers/trainer.py

with @ArthurZucker's suggestion

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>
